MapboxNavigation/StyleManager.swift

```diff
@@ -212,7 +213,8 @@ extension Date {
             guard let sunriseDate = calendar.date(from: sunriseComponents) else {
                 return nil
             }
-            return sunriseDate.timeIntervalSince(date)
+            let interval = sunriseDate.timeIntervalSince(date)
+            return interval >= 0 ? interval : (interval + 24 * 3600)
```

Comment:
isn't `interval` measured in seconds already?

Reply:
Because the sunrise and sunset times calculated by Solar are for a given day.
When it's night-time but before midnight, we are actually using the sunrise time of the previous morning (morning of this day).
For example 20:30 on a given day if after 6:30, which leads to a time difference in the past, so negative.
On the other hand, if past midnight, 1:30 is still night time, but is before 6:30 of the same 24 hours day, which leads to a positive time difference.